### ISSUE TITLE
Disable depending on ring for x86, armv7 targets

### ISSUE THREAD

**Issue by zonyitoo** [shadowsocks/shadowsocks-rust#347]
https://github.com/shadowsocks/shadowsocks-android/pull/2622
https://github.com/shadowsocks/crypto2/issues/6

1. Set `RUSTFLAGS` separately for `x86`

https://github.com/shadowsocks/shadowsocks-rust/blob/d0fc26317d5ce26de9dafe4dd1c241a9f23b1cfc/.cargo/config.toml#L6-L11

2. Disable `ring` for `x86` and `armv7` compile target

https://github.com/shadowsocks/shadowsocks-rust/blob/d0fc26317d5ce26de9dafe4dd1c241a9f23b1cfc/shadowsocks/Cargo.toml#L90-L94

@madeye Please verify if these modification works for your x86 Android device.

**Comment by madeye** [comment]
Yes, with the following change, shadowsocks-rust works now on x86 android simulator.

--- a/.cargo/config.toml
+++ b/.cargo/config.toml
```diff
@@ -3,7 +3,14 @@ rustflags = [
     # "--print", "native-static-libs",
 ]

-[target.'cfg(any(target_arch = "x86", target_arch = "x86_64"))']
+[target.'cfg(any(target_arch = "x86"))']
+rustflags = [
+    # "--print", "native-static-libs",
+    # "-C", "target-cpu=native",
+    "-C", "target-feature=+aes,+pclmul,+sse,+sse2,+sse3,+ssse3",
+]
+
+[target.'cfg(any(target_arch = "x86_64"))']
 rustflags = [
     # "--print", "native-static-libs",
     # "-C", "target-cpu=native",
```
--- a/shadowsocks/Cargo.toml
+++ b/shadowsocks/Cargo.toml

```diff
@@ -87,10 +87,16 @@ tower = { git = "https://github.com/tower-rs/tower.git", optional = true }
 ipnet = "2.3"
 iprange = "0.6"

-[target.'cfg(any(target_arch = "x86", target_arch = "x86_64"))'.dependencies]
+[target.'cfg(any(target_arch = "arm"))'.dependencies]
+shadowsocks-crypto = { version = "0.1" }
+
+[target.'cfg(any(target_arch = "x86"))'.dependencies]
+shadowsocks-crypto = { version = "0.1" }
+
+[target.'cfg(any(target_arch = "x86_64"))'.dependencies]
 shadowsocks-crypto = { version = "0.1", features = ["ring"] }

-[target.'cfg(any(target_arch = "arm", target_arch = "aarch64"))'.dependencies]
+[target.'cfg(any(target_arch = "aarch64"))'.dependencies]
 shadowsocks-crypto = { version = "0.1", features = ["ring"] }

 [target.'cfg(not(any(target_arch = "x86", target_arch = "x86_64", target_arch = "arm", target_arch = "aarch64")))'.dependencies]
```

**Comment by zonyitoo** [comment]
Great. These changes will be included in v1.9.0 release.
